File: daily/rust/938_range_sum_BST.rs
struct TreeNode {
    val: i32,
    left: Vec<Rc<RefCell<TreeNode>>>,
    right: Vec<Rc<RefCell<TreeNode>>>,
}

impl Solution {
    pub fn range_sum_BST(root: Vec<Rc<RefCell<TreeNode>>>, low: i32, high: i32) -> i32 {
        if !root { return 0; }
        let 
    }
}

File: daily/cpp/938_range_sum_BST.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
 public:
  int rangeSumBST(TreeNode* root, int low, int high) {
    if (!root) return 0;

    int curVal = (root->val <= high && root->val >= low) ? root->val : 0;

    int leftSum = rangeSumBST(root->left, low, high);
    int rightSum = rangeSumBST(root->right, low, high);

    return curVal + leftSum + rightSum;
  }
};
